homeassistant/components/mqtt/__init__.py

```diff
@@ -1156,9 +1159,14 @@ def available(self) -> bool:
 class MqttDiscoveryUpdate(Entity):
     """Mixin used to handle updated discovery message."""
 
-    def __init__(self, discovery_hash, discovery_update=None) -> None:
+    def __init__(self, discovery_data, discovery_update=None) -> None:
         """Initialize the discovery update mixin."""
-        self._discovery_hash = discovery_hash
+        self._discovery_hash = (
+            discovery_data[ATTR_DISCOVERY_HASH] if discovery_data else None
+        )
+        self._discovery_topic = (
```

Comment:
Why keep two variables around if you can just store `discovery_data`  ?

Reply:
You're right, cleaned up.